src/int.rs

```diff
@@ -558,6 +616,93 @@ impl ops::Not for I256 {
     }
 }
 
+impl ops::BitAnd for I256 {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        I256(self.0 & rhs.0)
+    }
+}
+
+impl ops::BitAndAssign for I256 {
+    fn bitand_assign(&mut self, rhs: Self) {
+        *self = *self & rhs;
+    }
+}
+
+impl ops::BitOr for I256 {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self::Output {
+        I256(self.0 | rhs.0)
+    }
+}
+
+impl ops::BitOrAssign for I256 {
+    fn bitor_assign(&mut self, rhs: Self) {
+        *self = *self | rhs;
+    }
+}
+
+impl ops::BitXor for I256 {
+    type Output = Self;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        I256(self.0 ^ rhs.0)
+    }
+}
+
+impl ops::BitXorAssign for I256 {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        *self = *self ^ rhs;
+    }
+}
+
+macro_rules! impl_shift {
+    ($( $t:ty $( [ $convert:ident ] )? ),*) => {
+        $(
+            impl_shift!(__impl $t $([$convert])*);
+        )*
+    };
+    (__impl $t:ty) => {
+        impl_shift!(__impl $t [ from ]);
+    };
+    (__impl $t:ty [ $convert:ident ]) => {
+        impl ops::Shl<$t> for I256 {
+            type Output = Self;
+
+            fn shl(self, rhs: $t) -> Self::Output {
+                // NOTE: We are OK with wrapping behaviour here, that is how
+                //   Rust behaves with the primitive integer types.
+                I256(self.0 << I256::$convert(rhs).0)
+            }
+        }
+
+        impl ops::ShlAssign<$t> for I256 {
+            fn shl_assign(&mut self, rhs: $t) {
+                *self = *self << rhs;
+            }
+        }
+
+        impl ops::Shr<$t> for I256 {
+            type Output = Self;
+
+            fn shr(self, rhs: $t) -> Self::Output {
+                I256(self.0 >> I256::$convert(rhs).0)
+            }
+        }
+
+        impl ops::ShrAssign<$t> for I256 {
+            fn shr_assign(&mut self, rhs: $t) {
+                *self = *self >> rhs;
+            }
+        }
+    };
+}
+
+impl_shift!(i8, u8, i16, u16, i32, u32, i64, u64, i128, u128, isize, usize);
+impl_shift!(I256, U256[from_raw]);
```

Comment:
what does `U256[from_raw]` mean?

Reply:
Its because `I256: !From<U256>` because of potential overflows. I handle the special syntax in the macro here: https://github.com/gnosis/ethcontract-rs/pull/226/files/9cd36719f8a02d50381bc76c637e1896a1f90b12#diff-dce46b69f16e6755634e0bc166dcc3ecR667-R670

By default, types will use `from` conversion, but `U256` needs to use `from_raw` conversion.